keep/actions/actions_factory.py

```diff
@@ -0,0 +1,82 @@
+import time
+import logging
+import yaml
+from io import StringIO
+from uuid import uuid4
+from typing import List, Union
+from keep.api.models.action import ActionDTO
+from keep.api.models.db.action import Action
+from keep.api.core.db import get_all_actions, create_actions, delete_action, get_action, update_action
+from keep.actions.actions_exception import ActionsFactoryException
+
+logger = logging.getLogger(__name__)
+
+class ActionsFactory:
+    """CRUD for Action model that shares across CLI, API, ..."""
+    @staticmethod
+    def get_all_actions(tenant_id: str) -> List[ActionDTO]:
+        action_models = get_all_actions(tenant_id)
+        action_dtos = map(ActionsFactory._convert_model_to_dto, action_models)
+        return list(action_dtos)
+
+    @staticmethod
+    def _convert_model_to_dto(model: Action) -> ActionDTO:
+        return ActionDTO(
+            id=model.id, use=model.use, name=model.name, details=yaml.safe_load(StringIO(model.action_raw))
+        )
+
+    @staticmethod
+    def add_actions(tenant_id: str, installed_by: str, action_dtos: List[dict]):
+        try:
+            actions = map(lambda dto: Action(
+                id=str(uuid4()),
+                tenant_id=tenant_id,
+                installed_by=installed_by,
+                installation_time=time.time(),
+                name=dto.get("name"),
+                use=dto.get("use") or dto.get("name"),
+                action_raw=yaml.dump(dto)
+            ), action_dtos)
+            create_actions(actions)
+        except Exception as e:
+            logger.exception("Failed to create actions", extra={ "error": e })
```

Comment:
this is redundant since logger.exception already logs the exception stacktrace

Reply:
It's revised to remove extra error information. 

```
logger.exception("Failed to create actions")
```